crates/astria-conductor/src/executor.rs

```diff
@@ -306,6 +341,34 @@ impl<C: ExecutionClient> Executor<C> {
             .remove(sequencer_block_hash);
         Ok(())
     }
+
+    /// This function attempts to execute all blocks in the block queue that have been recieved
+    /// without their parent block being executed yet. It always called when a block is recieved
+    /// from the gossip network, after execute_block has been called on the newly received block.
+    ///
+    /// # Errors
+    ///
+    /// This function returns an error if:
+    /// - The call to execute_block fails
+    async fn try_execute_queue(&mut self) -> Result<()> {
+        while let Some((block, _)) = self.block_queue.clone().into_sorted_iter().last() {
+            if block
+                .header
+                .last_block_id
+                .expect("Could not unwrap last block")
```

Comment:
we should probably check this before putting it into the queue and remove the `expect` here

also, `last_block_id` should always be set except in the case of the genesis block, which we shouldn't be receiving in the conductor anyways

Reply:
i made a `is_next_block()` function that checks if the id exists and must be passed before adding to the queue